src/App.tsx

```diff
@@ -1,14 +1,95 @@
 /* eslint-disable max-len */
-import React from 'react';
+// import React, { useCallback, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+const filteredTodos = (todos:Todo[], query:string, filterBy: string) => {
+  let preparedTodos = [...todos];
+
+  if (query) {
+    const preparedQuery = query.trim().toLowerCase();
+
+    preparedTodos = preparedTodos.filter(todo => todo.title.toLowerCase().includes(preparedQuery));
+  }
+
+  switch (filterBy) {
+    case 'active':
+      preparedTodos = preparedTodos.filter(todo => !todo.completed);
+      break;
+    case 'completed':
+      preparedTodos = preparedTodos.filter(todo => todo.completed);
+      break;
+    case 'all':
+    default:
+      preparedTodos = [...preparedTodos];
+      break;
+  }
+
+  return preparedTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState('');
+
+  // let visibleTodos = [...todos];
+
+  // if (query) {
+  //   const preparedQuery = query.trim().toLowerCase();
+
+  //   visibleTodos = visibleTodos.filter(todo => todo.title.toLowerCase().includes(preparedQuery));
+  // }
+
+  // switch (filterBy) {
+  //   case 'active':
+  //     visibleTodos = visibleTodos.filter(todo => !todo.completed);
+  //     break;
+  //   case 'completed':
+  //     visibleTodos = visibleTodos.filter(todo => todo.completed);
+  //     break;
+  //   case 'all':
+  //   default:
+  //     visibleTodos = [...visibleTodos];
+  //     break;
+  // }
+  const visibleTodos = filteredTodos(todos, query, filterBy);
+
+  const fetchData = async () => {
+    try {
+      const data = await getTodos();
+
+      setTodos(data);
+    } catch (error) {
+      if (error instanceof Error) {
+        alert(error.message);
```

Comment:
will be better to handle errors with some message for the user on the page instead of using `alert`